pkg/apis/extensions/validation/validation.go

```diff
@@ -564,3 +569,30 @@ func ValidateClusterAutoscaler(autoscaler *extensions.ClusterAutoscaler) errs.Va
 	allErrs = append(allErrs, validateClusterAutoscalerSpec(autoscaler.Spec)...)
 	return allErrs
 }
+
+func ValidatePodSelector(ps extensions.PodSelector) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateLabels(ps.MatchLabels, "matchLabels")...)
+	for i, expr := range ps.MatchExpressions {
```

Comment:
I agree with @bgrant0607 but lack a good example now.  I don't think allowing it makes it necessarily harder to analyze.